client/extensions/woocommerce/state/sites/payment-methods/actions.js

```diff
@@ -107,3 +100,47 @@ export const savePaymentMethod = ( siteId, method, successAction = null, failure
 			}
 		} );
 };
+
+const savePaymentMethodEnabledSuccess = ( siteId, data ) => {
+	const paymentMethod = addPaymentMethodDetails( data );
+	return {
+		type: WOOCOMMERCE_PAYMENT_METHOD_ENABLED_UPDATE_SUCCESS,
+		siteId,
+		data: paymentMethod,
+	};
+};
+
+export const savePaymentMethodEnabled = (
+	siteId,
+	method,
+	enabled,
+	successAction = null,
+	failureAction = null
+) => ( dispatch, getState ) => {
+	const state = getState();
+	if ( ! siteId ) {
+		siteId = getSelectedSiteId( state );
+	}
+	const updateAction = {
+		type: WOOCOMMERCE_PAYMENT_METHOD_ENABLED_UPDATE,
+		siteId,
+	};
+
+	const body = { settings: { enabled } };
+
+	dispatch( updateAction );
+
+	return request( siteId ).put( `payment_gateways/${ method.id }`, body )
+		.then( ( data ) => {
+			dispatch( savePaymentMethodEnabledSuccess( siteId, data ) );
+			if ( successAction ) {
+				dispatch( successAction( data ) );
+			}
+		} )
+		.catch( err => {
+			dispatch( setError( siteId, updateAction, err ) );
```

Comment:
I think we could use more details in `updateAction` to make debugging easier if this breaks. The 2nd parameter to setError is meant to give context to the error - https://github.com/Automattic/wp-calypso/blob/b564f310aff54fdd9be406254349a1e28d3ee371/client/extensions/woocommerce/state/sites/status/wc-api/actions.js#L13

See how I populated `postAction` in https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/state/sites/setup-choices/actions.js#L47 for an example

Consider adding the method.id and enabled state